### PR TITLE
Initial MatchManager implementation

### PR DESCRIPTION
**Pull request by HishamYahya**
This PR adds the MatchManager class, which orchestrates the matching between a given list of any number of LMs, keeps track of the wins/draws/losses, and records the ELOs of each model in the list.

For now, the only matching strategy implemented is Round-Robin.

(HK) edited

### CODE REVIEW

File: zero_sum_eval/managers/match_manager.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        if self.pair_index >= len(self.matches):
+            # Reset matches
+            self.pair_index = 0
+            self.round += 1
+
+        next_match = self.matches[self.pair_index]
+        self.pair_index += 1
+        return next_match
+
+class MatchManager:
```
Comment (haidark) [inline zero_sum_eval/managers/match_manager.py:56]: This class should use GameManager to run the individual games in the match.

Comment (haidark) [reply zero_sum_eval/managers/match_manager.py:56]: @HishamYahya

**Comment by HishamYahya** [comment]
@haidark ready for review

File: tournament_main.py
Comment (haidark) [inline tournament_main.py:1]: It looks like this file should be deleted (replaced by `run_matches.py`

File: zero_sum_eval/games/chess/chess_game.py
Comment (haidark) [inline zero_sum_eval/games/chess/chess_game.py:1]: Are these methods used anywhere? I think they might be left over from some previous thinking. 
The `GameState.validate_game()` method should handle this functionality

**Review by haidark — COMMENTED** [review]
Looks good to me, great to see GameManager being recycled here.

I will push the two small changes (deletions) and merge,

**Pull request merged**